### PR TITLE
CT-649 MtoA .ass dependency scraping (first pass)

### PR DESCRIPTION
**Pull request by lawschlosser**

### CODE REVIEW

File: conductor/lib/maya_utils.py [hunk 1]
```diff
@@ -519,13 +523,41 @@ def collect_dependencies(node_attrs):
...
                     dependencies.append(path)
 
     # ---- OCIO SCRAPING -----
     ocio_dependencies = scrape_ocio_dependencies()
     logger.debug("ocio_dependencies: %s", ocio_dependencies)
     dependencies.extend(ocio_dependencies)
 
+    # ---- ARNOLD .ASS SCRAPING -----
+    # If any ass_filepaths were found, we need to process their dependencies as well.
+    # Note that we intentially maintain an explicit list of .ass files (rather than simply searching
```
Comment (flebel) [inline conductor/lib/maya_utils.py]: `s/intentially/intentionally/`

File: conductor/lib/maya_utils.py [hunk 2]
```diff
@@ -519,13 +523,41 @@ def collect_dependencies(node_attrs):
...
+    # through all of our dependencies thus far for a .ass extension) because there may some ass files
+    # that are dependecies, but are not necessary (or redundant) to load/parse (i.e. in the case of
```
Comment (flebel) [inline conductor/lib/maya_utils.py]: `s/dependecies/dependencies/`

File: conductor/lib/maya_utils.py [hunk 3]
```diff
@@ -910,6 +942,106 @@ def is_arnold_tx_enabled():
...
+    Read/load the given arnold files with the arnold api, and seek out nodes that have filepaths
+    of interest.
+
+    todo(lws): Still need to investigate why this crashes in mayapy when arnold plugins are
+        present (such as yeti or alshaders).
+
+    node_attrs: dictionary. key is the node type, value is a list of node attributes to query.
+
+    plugin_paths: tuple of strings. This may be necessary when running outside of maya/interactive.
+        so that noes for arnold plugins (yeti, etc) can be properly loaded/used.
```
Comment (flebel) [inline conductor/lib/maya_utils.py]: `noes`?

File: conductor/lib/maya_utils.py [hunk 4]
```diff
@@ -910,6 +942,106 @@ def is_arnold_tx_enabled():
...
+
+    arnold.AiBegin()
+    arnold.AiMsgSetConsoleFlags(arnold.AI_LOG_ALL)
+    if plugin_paths:
+        arnold.AiLoadPlugins(os.path.sep.join(plugin_paths))
+    paths = []
+    for ass_filepath in ass_filepaths:
+        deps = _scrape_ass_file(ass_filepath, node_attrs)
+        paths.extend(deps)
+    arnold.AiEnd()
```
Comment (flebel) [inline conductor/lib/maya_utils.py]: Should this be wrapped in a `try-finally` block?

File: conductor/lib/maya_utils.py [hunk 5]
```diff
@@ -910,6 +942,106 @@ def is_arnold_tx_enabled():
...
+    Read/load the given arnold file with the arnold api, and seek out nodes that have filepaths
+    of interest.
+
+    todo(lws): Still need to investigate why this crashes in mayapy when arnold plugins are 
+        present (such as yeti or alshaders).
+
+    node_attrs: dictionary. key is the node type, value is a list of node attributes to query.
+
+    plugin_paths: tuple of strings. This may be necessary when running outside of maya/interactive.
+        so that noes for arnold plugins (yeti, etc) can be properly loaded/used.
```
Comment (flebel) [inline conductor/lib/maya_utils.py]: `noes`?

File: conductor/lib/maya_utils.py (hunk 4 above)
Comment (lawschlosser) [reply conductor/lib/maya_utils.py]: good call

**Pull request merged**